ui/index.d.ts

```diff
@@ -203,6 +203,13 @@ export declare class AzureActionHandler {
     registerEvent<T>(eventId: string, event: Event<T>, callback: (this: IActionContext, ...args: any[]) => any): void;
 }
 
+/**
+ * Runs the given code. If it succeeds or is canceled, nothing else happens. If it fails, telemetry is sent (with id "UnexpectedError"),
+ * and the error is also sent to the outputChannel if defined.
+ * @param errorContext A string describing the context of the code in case of an error.
+ */
+export declare function callWithUnxpectedErrorTelemetry(errorContext: string, telemetryReporter: TelemetryReporter | undefined, outputChannel: OutputChannel | undefined, callback: (this: IActionContext) => any, extensionContext: ExtensionContext): Promise<any>;
```

Comment:
So this goes outside our established pattern for error telemetry and I think I'd need more explanation to get on board.

For example, one of the biggest ways I prioritize error telemetry is _percentage_ of failed events. If you never report 'succeeded' or 'canceled', then there is no percentage. Even if you look at raw number of failed events (instead of percentage), you still have all of the events lumped into 'UnexpectedError'. We could modify our query to use 'errorContext' instead of the propertyName, but I just don't see the value in changing the norm